aws/resource_aws_opsworks_application.go

```diff
@@ -77,15 +69,9 @@ func resourceAwsOpsworksApplication() *schema.Resource {
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"type": {
-							Type:     schema.TypeString,
-							Required: true,
-							ValidateFunc: validation.StringInSlice([]string{
-								opsworks.SourceTypeGit,
-								opsworks.SourceTypeSvn,
-								opsworks.SourceTypeArchive,
-								opsworks.SourceTypeS3,
-								"other",
-							}, false),
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(opsworks.SourceType_Values(), false),
```

Comment:
It looks like we need to include the (missing enum `other`) still:

```
=== CONT  TestAccAWSOpsworksApplication_basic
TestAccAWSOpsworksApplication_basic: resource_aws_opsworks_application_test.go:22: Step 1/3 error: terraform failed: exit status 1
stderr:
Error: expected app_source.0.type to be one of [git svn archive s3], got other
on config312294631/terraform_plugin_test.tf line 118, in resource "aws_opsworks_application" "test":
118:     type = "other"
--- FAIL: TestAccAWSOpsworksApplication_basic (4.38s)
```

```suggestion
							ValidateFunc: validation.StringInSlice(append(opsworks.SourceType_Values(), "other"), false),
```

Reply:
ahh so close 😅